smithy-trait-codegen/src/main/java/software/amazon/smithy/traitcodegen/writer/TraitCodegenWriter.java

```diff
@@ -242,6 +232,32 @@ private String getPlaceholder(Symbol symbol) {
             // Return a placeholder value that will be filled when toString is called
             return format("$${$L:L}", normalizedSymbol.getFullName());
         }
+
+        private boolean isGenericType(Symbol symbol) {
+            String name = symbol.getName();
+            return name.equals("List") || name.equals("Map")
+                    || name.equals("Set")
+                    || name.equals("Collection");
+        }
```

Comment:
These all seem to be somewhat of a Collection. Might want to call it `isCollectionTypeName` since it represents that symbol is a type name such as the word `List` , `Map` , `Set` , etc. 
```suggestion

         // At top level in the class 
         private static Set<String> COLLECTION_TYPE_NAMES = SetUtils.of("List", "Map", "Set", "Collection");
         
        private boolean isGenericType(Symbol symbol) {
            return COLLECTION_TYPE_NAMES.contains(symbol.getName());
        }
```

It can also be inlined to where this is used and the method can be removed in it's entirety. This prevents creation one by one checking for each symbol and also has a single instantiation of the set which can be re-used to look up the matching symbol efficiently.

Reply:
I also like the name `parametrized` as used by Manuel in his comment.